airflow/contrib/hooks/gcp_vision_hook.py

```diff
@@ -0,0 +1,245 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import re
+
+from google.api_core.exceptions import AlreadyExists, GoogleAPICallError, RetryError
+from google.cloud.vision_v1 import ProductSearchClient
+from google.cloud.vision_v1.proto.product_search_service_pb2 import Product
+from google.protobuf.json_format import MessageToJson
+
+from airflow import AirflowException
+from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
+
+PARENT_TEMPLATE = 'projects/{}/locations/{}'
```

Comment:
I think, I think you should use 
https://googleapis.github.io/google-cloud-python/latest/vision/gapic/v1/api.html#google.cloud.vision_v1.ProductSearchClient.location_path
https://googleapis.github.io/google-cloud-python/latest/vision/gapic/v1/api.html#google.cloud.vision_v1.ProductSearchClient.product_path
https://googleapis.github.io/google-cloud-python/latest/vision/gapic/v1/api.html#google.cloud.vision_v1.ProductSearchClient.product_set_path

Reply:
Thanks, good call. I've implemented it in https://github.com/apache/airflow/pull/4665